src/app/grid/grid-save-state/state.directive.ts

```diff
@@ -215,5 +209,23 @@ export class IgxGridStateDirective implements AfterViewInit {
         });
         return gridColumns;
     }
+
+    private stringifyCallback(key: string, val: any) {
+        const expr = val as IFilteringExpression;
+        if (expr && expr.searchVal instanceof Set) {
+            expr.searchVal = Array.from(expr.searchVal);
+            return expr;
+        }
```

Comment:
Instead of modifying the expression, you should check the `key` and `val` parameters of the function:
```ts
        if (key === "searchVal" && val instanceof Set) {
            return Array.from(val);
        }
```

The same should be applied to the parseCallback too.

Reply:
Done

Reply:
Done